### ISSUE TITLE
False positive in conditional compilation

### ISSUE THREAD

**Issue by KacperCzapp** [realm/SwiftLint#5223]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When code is wrapped in a conditional compilation block (`#if DEBUG`, `#if os(tvOS)` etc.), the `implicit_return` rule is triggered even though it's not applicable and would even break the code.

```swift
#if os(iOS)
var isSelected: Bool = false {
    didSet {
        guard isSelected else { return } // triggered here
        actionButtonTapped()
    }
}
#endif
```
```swift
#if os(tvOS)
override var preferredFocusEnvironments: [UIFocusEnvironment] {
    guard isEnabled else {
        return super.preferredFocusEnvironments // triggered here
    }
    return [someView]
}
#endif
```
```swift
#if os(tvOS)
override var preferredFocusEnvironments: [UIFocusEnvironment] {
    if isEnabled {
        return [someView] // triggered here
    } else {
        return super.preferredFocusEnvironments // triggered here
    }
}
#endif
```

### Environment

* SwiftLint version: **0.52.4**
* Installation method: **Homebrew**
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **No**
* Which Xcode version are you using? **Xcode 14.3.1 (14E300c)**

**Comment by SimplyDanny** [comment]
These cases have meanwhile been fixed in #5166 which will become active in the upcoming release.
